projects/Mallard/src/navigation/index.ts

```diff
@@ -157,9 +156,11 @@ const RootNavigator = createAppContainer(
                     }: {
                         navigation: NavigationScreenProp<{}>
                     }) => {
-                        const settings = useOtherSettingsValues()
+                        const shouldShowOnboarding = useShouldShowOnboarding()
                         useEffect(() => {
-                            if (shouldShowOnboarding(settings)) {
+                            /** Setting is still loading, do nothing yet. */
+                            if (shouldShowOnboarding == null) return
```

Comment:
I'm slightly worried about having some of these return `null` when expecting a boolean, it leave a bit of a gap where people may just treat it as a boolean and forget that it will be null for a period. I wonder if we can have a more solid result type along the lines of `pending`, `success` and `error`?

Reply:
Agreed. Let's switch to algebraic-type model.

Reply:
👍